src/cider/type/plan/UnaryExpr.cpp

```diff
@@ -37,36 +37,83 @@ JITExprValue& UOper::codegen(JITFunction& func) {
     CIDER_THROW(CiderCompileException,
                 "Decimal and TimeInterval are not supported in Uoper codegen now.");
   }
-  FixSizeJITExprValue operand_val(operand->codegen(func));
-  const auto& ti = get_type_info();
-  const auto type = ti.is_decimal() ? decimal_to_int_type(ti) : ti.get_type();
+
   switch (get_optype()) {
-    case kISNULL: {
-      // For ISNULL, the null will always be false
-      auto null_value = operand_val.getNull().get()
-                            ? operand_val.getNull()
-                            : func.createConstant(getJITTag(type), false);
-      return set_expr_value(func.createConstant(getJITTag(type), false), null_value);
-    }
+    case kISNULL:
     case kISNOTNULL: {
-      auto null_value = operand_val.getNull().get()
-                            ? operand_val.getNull()->notOp()
-                            : func.createConstant(getJITTag(type), true);
-      return set_expr_value(func.createConstant(getJITTag(type), false), null_value);
+      return codegenIsNull(func, operand, get_optype() == kISNOTNULL);
     }
     case kNOT: {
-      CHECK(operand_val.getNull().get());
-      return set_expr_value(operand_val.getNull(), operand_val.getValue()->notOp());
+      return codegenNot(func, operand);
     }
     case kCAST: {
-      return set_expr_value(operand_val.getNull(),
-                            codegenCastFunc(func, operand_val.getValue()));
+      return codegenCast(func, operand);
     }
     default:
       UNIMPLEMENTED();
   }
 }
 
+JITExprValue& UOper::codegenIsNull(JITFunction& func,
+                                   Analyzer::Expr* operand,
+                                   bool use_is_not_null) {
+  // For ISNULL, the null will always be false
+  const auto& ti = get_type_info();
+  const auto type = ti.is_decimal() ? decimal_to_int_type(ti) : ti.get_type();
+
+  // for IS NULL / IS NOT NULL, we only need the null info (getNull())
+  JITExprValueAdaptor operand_val(operand->codegen(func));
+
+  if (use_is_not_null) {
+    // is not null
+    auto null_value = operand_val.getNull().get()
+                          ? operand_val.getNull()->notOp()
```

Comment:
seems `operand_val.getNull().get()` always exists even for non-nullable columns. Logic here need updated.`null_value=operand_val.getNull()->notOp()`.